staging/src/k8s.io/kubectl/pkg/util/completion/completion.go

```diff
@@ -128,15 +130,16 @@ func ResourceAndPortCompletionFunc(f cmdutil.Factory) func(*cobra.Command, []str
 // 1- pod names that match the toComplete prefix
 // 2- resource types containing pods which match the toComplete prefix
 // and as a second argument the containers within the specified pod.
-func PodResourceNameAndContainerCompletionFunc(f cmdutil.Factory) func(*cobra.Command, []string, string) ([]string, cobra.ShellCompDirective) {
+func PodResourceNameAndContainerCompletionFunc(restClientGetter genericclioptions.RESTClientGetter) func(*cobra.Command, []string, string) ([]string, cobra.ShellCompDirective) {
 	return func(_ *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
 		var comps []string
 		directive := cobra.ShellCompDirectiveNoFileComp
 		if len(args) == 0 {
-			comps, directive = doPodResourceCompletion(f, toComplete)
+			comps, directive = doPodResourceCompletion(restClientGetter, toComplete)
 		} else if len(args) == 1 {
-			podName := convertResourceNameToPodName(f, args[0])
-			comps = CompGetContainers(f, podName, toComplete)
+			builder := resource.NewBuilder(restClientGetter)
```

Comment:
We are instantiating new builder from restClientGetter. Because we can't use the builder in `Factory`.